packages/sw-precaching/src/lib/install-handler.js

```diff
@@ -13,49 +13,69 @@
  limitations under the License.
 */
 
-import assert from '../../../../lib/assert';
-import {version, defaultCacheId} from './constants';
+import ErrorFactory from './error-factory';
+import IDBHelper from '../../../../lib/idb-helper.js';
+import {version, defaultCacheId, DB_NAME, DB_VERSION, DB_STORENAME}
+  from './constants';
 
-const setOfCachedUrls = (cache) => {
-  return cache.keys()
-    .then((requests) => requests.map((request) => request.url))
-    .then((urls) => new Set(urls));
+const getPreviousRevisionDetails = (idbHelper, path) => {
+  return idbHelper.get(path);
 };
 
-const urlsToCacheKeys = (precacheConfig) => new Map(
-  /** urls.map(item => {
-    var relativeUrl = item[0];
-    var hash = item[1];
-    var absoluteUrl = new URL(relativeUrl, self.location);
-    var cacheKey = createCacheKey(absoluteUrl, hashParamName, hash, /a/);
-    return [absoluteUrl.toString(), cacheKey];
-  })**/
-);
-
-export default ({assetsAndHahes, cacheId} = {}) => {
-  assert.isType(assetsAndHahes, 'array');
-
-  self.addEventListener('install', (event) => {
-    const cacheName =
-      `${cacheId || defaultCacheId}-${version}-${self.registration.scope}`;
-
-    event.waitUntil(
-      caches.open(cacheName).then((cache) => {
-        return setOfCachedUrls(cache).then((cachedUrls) => {
-          return Promise.all(
-            Array.from(urlsToCacheKeys.values()).map(function(cacheKey) {
-              // If we don't have a key matching url in the cache already,
-              // add it.
-              if (!cachedUrls.has(cacheKey)) {
-                return cache.add(
-                  new Request(cacheKey, {credentials: 'same-origin'}));
-              }
-
-              return Promise.resolve();
-            })
-          );
-        });
-      })
-    );
+const addAssetHashToIndexedDB = (idbHelper, assetAndHash) => {
+  return idbHelper.put(assetAndHash.path, assetAndHash.revision);
+};
+
+const installHandler = ({assetsAndHahes, cacheId} = {}) => {
+  if (!Array.isArray(assetsAndHahes)) {
+    throw ErrorFactory.createError('assets-not-an-array');
+  }
+
+  const cacheName =
```

Comment:
One other comment. If the developer adds the library to the window referencing the scope at this level will throw and error.